File: src/main/java/com/metjka/vort/ui/GlobalContextMenu.java
package com.metjka.vort.ui;

import javafx.scene.control.ContextMenu;
import javafx.scene.control.MenuItem;

/**
 * A context menu with global actions (i.e. quit).
 */
public class GlobalContextMenu extends ContextMenu {
    public GlobalContextMenu(MenuActions menuActions) {
        super();


        MenuItem menuFullScreen = new MenuItem("Toggle full screen");
        menuFullScreen.setOnAction(menuActions::toggleFullScreen);
        
        MenuItem menuQuit = new MenuItem("Quit");
        menuQuit.setOnAction(menuActions::onQuit);

    }
}

File: src/main/java/io/metjka/vortex/ui/GlobalContextMenu.kt
package io.metjka.vortex.ui

import javafx.event.ActionEvent
import javafx.event.EventHandler
import javafx.scene.control.ContextMenu
import javafx.scene.control.MenuItem

/**
 * A context menu with global actions (i.e. quit).
 */
class GlobalContextMenu(menuActions: MenuActions) : ContextMenu() {
    init {


        val menuFullScreen = MenuItem("Toggle full screen")
        menuFullScreen.onAction = EventHandler { actionEvent -> menuActions.toggleFullScreen(actionEvent) }

        val menuQuit = MenuItem("Quit")
        menuQuit.onAction = EventHandler { actionEvent -> menuActions.onQuit(actionEvent) }

    }
}
